test_gpios.py

```diff
@@ -23,10 +25,25 @@ async def asyncSetUp(self):
             self.robot = await RobotClient.at_address(conf.address, opts)
         except ConnectionError:
             # There's some race condition in the Python SDK that causes
-            # reconnection to fail sometimes.
+            # reconnection to fail sometimes. See if we can figure out what
+            # the RDK server is doing that causes this trouble. Send a SIGUSR1
+            # (signal 10), which should log stack traces from all goroutines.
+            # The next tricky part here is that there might be 2 RDK servers
+            # running: the "normal" one on port 8080 and the board canary one
+            # on port 9090. We want to only send the signal to the latter. We
+            # achieve that by finding all processes, filtering for the ones
+            # that look like RDK servers, filtering for the ones that mention
+            # the canary config, and then removing all output from `ps` except
+            # the process ID.
+            subprocess.run("ps aux --cols 1000|" +
+                           "grep viam-server |" +
+                           "grep viam-canary.json |" +
+                           "sed 's/^[^ ]* *//g' |" +
+                           "sed 's/ .*$//g' |" +
+                           "xargs kill -10", shell=True)
```

Comment:
Oh, you could do `pkill -f` to match on the full invocation. E.g:
```
pkill -SIGQUIT -f viam-canary.json
```

Reply:
:facepalm: I couldn't find `-f`. Thanks!